drivers/phoenixcontact_modbus.c

```diff
@@ -100,23 +110,17 @@ void upsdrv_initinfo(void)
 	/* upsh.instcmd = instcmd; */
 	/* upsh.setvar = setvar; */
 
-	mrir(modbus_ctx, 0x0004, 1, &FWVersion);
-	dstate_setinfo("ups.firmware", "%" PRIu16, FWVersion);
-
-	mrir(modbus_ctx, 0x0005, 1, &PartNumber1);
-	mrir(modbus_ctx, 0x0006, 1, &PartNumber2);
-	mrir(modbus_ctx, 0x0007, 1, &PartNumber3);
-	mrir(modbus_ctx, 0x0008, 1, &PartNumber4);
+	mrir(modbus_ctx, 0x0005, 4, &tab_reg);
```

Comment:
Previously you were sending address of `uint16_t` to `mrir()` and now you send address of array, but the method signature seems not changed. Did you mean to read the register into just the zero'th element? Or did you drop the other 4 `mrir` lines by mistake?